Pytorch implementation

ailunyi opened the pull request:
1. Core changes
- Added PyTorch Implementation for both PennyLane gate-based and Orca photonic
- Changed quantum processing from one circuit to 4-channel grouped quantum processing (needs more testing to see if it is efficient)
- Added support for multiple quantum backends (PennyLane, PTLayer)

2. Questions/Review requested on... 
- Number of qubits and channels in the quantum layer needs some clarification
- I was using the basic ansatz with groups of 4 (as opposed to FQConv or HQConv in the literature), should I modify it to use a more advanced ansatz?

ailunyi committed:
added install requirements files, created utilsTorch and unetTorch, created demos for PyTorch for both centralized and federtated, quantum and classical

ailunyi committed:
created utils folder and deleted pycache

ailunyi committed:
deleted uneccessary file

ailunyi committed:
changed full quantum layer processing into batches of 4 for effiency, but need to do more testing. Fixed dependency issues, comments in code, and readme

bernalde commented on __pycache__/demo.cpython-312.pyc:
Add __pycache__ folder to .gitignore

bernalde commented on .gitignore:
```diff
@@ -0,0 +1,8 @@
+__pycache__/
+*.py[cod]
+results/
+data/
```
Add __pycache__ folder

bernalde commented on README.md:
````diff
@@ -0,0 +1,153 @@
+
+Our QVUNet architectures are inspired by the quantum hybrid diffusion models proposed in:
+
+```bibtex
````
We need to add more references to our previous work. In particular:

https://iopscience.iop.org/article/10.1088/2632-2153/ad8c11
https://pubs.acs.org/doi/10.1021/acs.iecr.4c03805 (and its corresponding repository)
https://openreview.net/forum?id=Is6AZWd0g5&referrer=%5Bthe%20profile%20of%20David%20E.%20Bernal%20Neira%5D(%2Fprofile%3Fid%3D~David_E._Bernal_Neira1)

bernalde reviewed (changes requested):
The requested changes are minor and primarily involve cleaning up the repository. Good work. I will activate a Copilot review, then we will evaluate how many of those changes will be implemented

ailunyi committed:
added additional references to previous work and fixed install reqirements

ailunyi committed:
1. added seeds for all demos (federated needs to be fixed) 2. fixed federated ray issues by replacing it with local imports & seperating conflixing JAX and Torch imports & moving functions from utils back into demo files  3. tried solving seed issues in federated rays but still needs to be worked on

ailunyi commented:
I have made a new commit, please review these changes! Thanks!
0. added new references to our past work & implemented copilot changes
1. added seeds for all demos (federated needs to be fixed) 
2. fixed federated ray issues by replacing it with local imports & separating conflicting JAX and Torch imports & moving functions from utils back into demo files 
3. tried solving seed issues in federated rays but still needs to be worked on

bernalde reviewed (approved):
Approved, good improvement. We merge it subject to copilot revision not indicating major flaws

bernalde committed:
Merge branch 'main' into pytorch-implementation

ailunyi committed:
implemented minor changes (syntax and print statement) recommended by copilot
